### PR TITLE
CSHARP-2764: Convert FLE examples from java doc to their c# equivalents.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        [Fact]
+        public void ClientSideEncryptionSimpleTour()
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            var localMasterKey = Convert.FromBase64String(LocalMasterKey); // 96 bytes
+
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+            var localKey = new Dictionary<string, object>
+                {
```
Comment (vincentkam) [inline tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs]: Nit: indentation

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+using MongoDB.Bson;
+using MongoDB.Driver.Encryption;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MongoDB.Driver.Examples
+{
+    public class ClientEncryptionExamples
+    {
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
```
Comment (rstam) [inline tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs:30]: Where did this value come from?

Are we leaking anybody's secret key?

**Comment by DmitryLukyanov** [comment]
Ready to review.

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                    }}
+                }},
+                'bsonType': 'object'
+            }}";
+            var autoEncryptionSettings = new AutoEncryptionOptions(
+                keyVaultNamespace,
+                kmsProviders,
+                schemaMap: new Dictionary<string, BsonDocument>()
+                {
+                    { collectionNamespace.ToString(), BsonDocument.Parse(schemaMap) }
```
Comment (rstam) [inline tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs:121]: Have you executed this code?

I'm concerned that `BsonDocument.Parse` might mess up the `keyId` value due to `GuidRepresentation` issues.

**Review by rstam — COMMENTED** [review]
No changes requested but I have a couple questions.

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs:30]: from the prose test: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs#L47

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs (hunk 3 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs:121]: >Have you executed this code?

yes, this code inserts the encrypted data(line `132`) in the `test.coll` and reads the decrypted data(line `134`).

NOTE: we use `BsonDocument.Parse` in the encryption prose test: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs#L344

Taking into account my above comment, do you think `BsonDocument.Parse` should be additionally checked?

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs (hunk 2 above)
Comment (rstam) [reply tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs:30]: OK

File: tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs (hunk 3 above)
Comment (rstam) [reply tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs:121]: If you have verified that we have no `GuidRepresentation` issues in this code then it's fine.

**Review by rstam — APPROVED** [review]

**Review by vincentkam — APPROVED** [review]
Thanks Dima!!!! LGTM! 😸

**Pull request closed without merging**